### ISSUE TITLE
Feature Request: Allow no required predict sets for a `Measure`

### ISSUE THREAD

**Issue by sebffischer** [mlr-org/mlr3#1094]
Some mlr3 Measures don't require a prediction object to operate on, such as `msr("selected_features")` or `msr("internal_valid_score")`. 

Let's say I want to resample xgboost with early stopping and I only care about the internal validation scores (e.g. in the context of tuning). I could then avoid making predictions on the test set alltogether. What I **want** to do is therefore:

```r
l = lrn("classif.xgboost", nrounds = 100,
  early_stopping_rounds = 10, validate = "test", predict_sets = NULL)
  
rr = resample(tsk("iris"), l, rsmp("holdout")

rr$aggregate(msr("internal_valid_score"))
```

The reason why the code above does not run is because each measure must have at least one required predict sets because of this check: https://github.com/mlr-org/mlr3/blob/3f159d91735b2b229605f34f35f22765749dd0c9/R/Measure.R#L147

I.e., for `MeasureInternalValidScore` I need to pick `"test"`, `"train"` or `"internal_valid"`, even though none are required.
The last call to `$aggregate()` will therefore err, no matter how I define the `MeasureInternalValidScore`.

**Comment by mllg** [comment]
You can toggle `empty.ok` and check if the unit tests still pass. But to be fair, not wanting to look at predictions at all is an unusual setting ;)
